Project/zepeto-modules/Assets/Zepeto Gesture Module/ZepetoScript/GestureLoader.ts

```diff
@@ -51,24 +55,66 @@ export default class GestureLoader extends ZepetoScriptBehaviour {
         newThumb.GetComponent<Thumbnail>().content = content;
 
         // Button Listener for each thumbnail
-        newThumb.GetComponent<Button>().onClick.AddListener(() => {
+        newThumb.GetComponent<Button>().onClick.AddListener(() => {  
             this.LoadAnimation(content);
         });
-        
         this.thumbnails.push(newThumb);
     }
 
     // 3. Loading Animation
-    private LoadAnimation(content: Content) {
+    private LoadAnimation(content: Content ) {
         // Verify animation load
         if (!content.IsDownloadedAnimation) {
             // If the animation has not been downloaded, download it.
             content.DownloadAnimation(() => {
                 // play animation clip
-                this._myCharacter.SetGesture(content.AnimationClip);
-            });
+                this.runAnimation(content.AnimationClip, content.Keywords)
+            });    
         } else {
-            this._myCharacter.SetGesture(content.AnimationClip);
+            this.runAnimation(content.AnimationClip, content.Keywords)
+        }
+    }
+       
+    // A function to run an animation, 
+    private runAnimation(animation: AnimationClip, type: OfficialContentType[] )
+    {        
+        //if there is another gesture running, stop the coroutine and cancel the gesture
+        if(this.gestureLoop)
+        {
+            this.StopCoroutine(this.gestureLoop)
+        }
+        this._myCharacter.CancelGesture()
+
+        // check if isLooping is true and it is not a pose
+        if(this.isLooping === true && this._checkIfGestureIsPose(type) === false)
+        {            
+            this.gestureLoop = this.StartCoroutine(this.setGestureLoop(animation))
+        }
+        //When the isLooping is false
```

Comment:
I think the code is self-explanatory and since we provided the commend for the `if` clause, this one can be deleted.

Reply:
Uhm, because I am using coroutine and I need to stop the coroutine in some other part of the code, this might be the best way to identify the coroutine when stopping it.

Reply:
Oh, I meant the comment.

```
        //When the isLooping is false
```